File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/Bug7721EntitySuperClass.java
package io.quarkus.it.panache.reactive;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.MappedSuperclass;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;

@MappedSuperclass
public class Bug7721EntitySuperClass extends PanacheEntity {

    @Column(nullable = false)
    public String superField = "default";

    public void setSuperField(String superField) {
        Objects.requireNonNull(superField);
        // should never be null
        Objects.requireNonNull(this.superField);
        this.superField = superField;
    }

}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/Bug7721EntitySuperClass.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass

@MappedSuperclass
open class Bug7721EntitySuperClass : PanacheEntity() {
    @JvmField
    @Column(nullable = false)
    var superField: String = "default"
}
